linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/CoreWasmLib.scala

```diff
@@ -2370,6 +2372,89 @@ final class CoreWasmLib(coreSpec: CoreSpec) {
     fb.buildAndAddToModule()
   }
 
+  /** `getSuperClass: (ref jlClass) -> (ref null jlClass)`.
+   *
+   *  This is the underlying func for the `Class_superClass` operation.
+   */
+  private def genGetSuperClass()(implicit ctx: WasmContext): Unit = {
+    val fb = newFunctionBuilder(genFunctionID.getSuperClass)
+    val jlClassParam = fb.addParam("jlClass", RefType(genTypeID.ClassStruct))
+    fb.setResultType(RefType.nullable(genTypeID.ClassStruct))
+
+    val typeDataLocal = fb.addLocal("typeData", RefType(genTypeID.typeData))
+    val kindLocal = fb.addLocal("kind", Int32)
+
+    // typeData := jlClass.data
+    fb += LocalGet(jlClassParam)
+    fb += StructGet(genTypeID.ClassStruct, genFieldID.classData)
+    fb += LocalTee(typeDataLocal)
+
+    // kind := typeData.kind
+    fb += StructGet(genTypeID.typeData, genFieldID.typeData.kind)
+    fb += LocalTee(kindLocal)
+
+    /* There are 3 cases that yield non-null results:
+     *
+     * - Scala classes that are not jl.Object (KindObject < kind <= KindClass)
+     * - JS types with a superClass (kind == KindJSTypeWithSuperClass)
+     * - Array classes (kind == KindArray)
+     *
+     * Note that KindArray < KindObject, and KindJSTypeWithSuperClass > KindObject,
+     * so we dispatch these two cases in the the two branches of an
```

Comment:
```suggestion
     * so we dispatch these two cases in the two branches of an
```

Also, by "those two cases", it means the first two items (`Scala classes that are not jl.Object` and `JS types with a superClass`, right?)

Reply:
Yes, that's right.